app/controllers/photos_controller.rb

```diff
@@ -60,4 +75,8 @@ def destroy
   def photo_params
     params.require(:photo).permit(:user_id, :category_id, :name, :desc, :upload)
   end
+
+  def has_permission?(photos)
+    Photo.user_id == current_user.id || current_user.admin?
+  end
```

Comment:
The argument coming in is photos. Photos is not used in the method call. Also Photo.user_id, what photo is that?